### PR TITLE
fix bugs of handler cleanupCache in TargetLoadPacking Plugin

### LINKED ISSUE #583: [Trimaran]Handler CacheCleanup will not clean up when all pods are expired
### Area

- [X] Scheduler
- [ ] Controller
- [ ] Helm Chart
- [ ] Documents

### Other components

_No response_

### What happened?

```
idx := sort.Search(len(cache), func(i int) bool {
			return cache[i].Timestamp.Add(metricsAgentReportingIntervalSeconds * time.Second).After(curTime)
		})
if idx == len(cache) {
	continue
}
```
When all pods are expired, the return value of idx is len(cache), No cleanup will be performed

### What did you expect to happen?

Clean up all outdated pods

### How can we reproduce it (as minimally and precisely as possible)?

_No response_

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Scheduler Plugins version

<details>

</details>

### PR DESCRIPTION
**Pull request by xiaomudk**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #583 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by ffromani** [comment]
LGTM considering the reported issue and how `sort.Search` works, but waiting review from @wangchen615 and @atantawi

File: pkg/trimaran/handler_test.go [hunk 1]
```diff
@@ -75,3 +76,71 @@ func TestHandlerCacheCleanup(t *testing.T) {
...
+
+func TestHandlerCacheCleanup(t *testing.T) {
+	now := time.Now()
+	tests := []struct {
+		name               string
+		scheduledPodsCache map[string][]podInfo
+		expectedCache      map[string][]podInfo
+	}{
+		{
+			// 没有过期pod
```
Comment (Huang-Wei) [inline pkg/trimaran/handler_test.go:88]: could you update the comments in English?

Comment (xiaomudk) [reply pkg/trimaran/handler_test.go]: @Huang-Wei  Already deleted

**Comment by Huang-Wei** [comment]
@xiaomudk could you rebase and I will request @atantawi to take another look.

**Pull request closed without merging**